server/src/parser/septicCnfg.ts

```diff
@@ -0,0 +1,324 @@
+            this.xvrRefs.get(ref.identifier)?.push(ref);
+        } else {
+            this.xvrRefs.set(ref.identifier, [ref]);
+        }
+    }
+}
+
+export class SepticBase {
+    start: number;
+    end: number;
+
+    constructor(start: number = -1, end: number = -1) {
+        this.start = start;
+        this.end = end;
+    }
+
+    updateEnd(): void {
+        return;
+    }
+}
+
+export class SepticObject extends SepticBase {
+    type: string;
+    identifier: Identifier | undefined;
+    attributes: Attribute[];
+
+    constructor(
+        type: string,
+        variable: Identifier | undefined,
+        start: number = -1,
+        end: number = -1
+    ) {
+        super(start, end);
+        this.type = type;
+        this.identifier = variable;
+        this.attributes = [];
+    }
+
+    addAttribute(attr: Attribute) {
+        this.attributes.push(attr);
+    }
+
+    updateEnd(): void {
+        this.identifier?.updateEnd();
+        this.attributes.forEach((elem) => {
+            elem.updateEnd();
+        });
+        if (this.attributes.length) {
+            this.end = this.attributes[this.attributes.length - 1].end;
+        }
+    }
+
+    getAttribute(name: string): Attribute | undefined {
+        return this.attributes.find((attr) => {
+            return attr.key === name;
+        });
+    }
+}
+
+export class Attribute extends SepticBase {
+    values: AttributeValue[];
+    key: string;
+
+    constructor(key: string, start: number = -1, end: number = -1) {
+        super(start, end);
+        this.values = [];
+        this.key = key;
+    }
+
+    addValue(value: AttributeValue) {
+        this.values.push(value);
+    }
+
+    updateEnd(): void {
+        if (this.values.length) {
+            this.end = this.values[this.values.length - 1].end;
+        }
+    }
+}
+
+export class Identifier extends SepticBase {
+    name: string;
+
+    constructor(name: string, start: number = -1, end: number = -1) {
+        super(start, end);
+        this.name = name.replace(/\s/g, "");
+    }
+}
+
+export class AttributeValue extends SepticBase {
+    value: string;
+    type: SepticTokenType;
+
+    constructor(
+        value: string,
+        type: SepticTokenType,
+        start: number = -1,
+        end: number = -1
+    ) {
+        super(start, end);
+        this.value = value;
+        this.type = type;
+    }
+}
+
+export function extractXvrRefs(obj: SepticObject): SepticReference[] {
+    const xvrRefs: SepticReference[] = [];
+    const regexVariable = /[a-zA-Z]+vr$/;
+    if (regexVariable.test(obj.type)) {
+        if (obj.identifier) {
+            let ref: SepticReference = {
+                identifier: obj.identifier.name,
+                location: {
+                    start: obj.identifier.start,
+                    end: obj.identifier.end,
+                },
+                obj: obj,
+            };
+            xvrRefs.push(ref);
+        }
+    }
+    switch (obj.type) {
```

Comment:
Would this be better to put in the yaml file?

Reply:
Updated to get this information from the yaml file.